cmd/thanos/query.go

```diff
@@ -536,3 +550,36 @@ func firstDuplicate(ss []string) string {
 
 	return ""
 }
+
+func engineFunc(newEngine func(time.Duration) *promql.Engine, lookbackDelta time.Duration, dynamicLookbackDelta bool) func(int64) *promql.Engine {
+	deltas := []int64{downsample.ResLevel0}
+	if dynamicLookbackDelta {
+		deltas = []int64{downsample.ResLevel0, downsample.ResLevel1, downsample.ResLevel2}
+	}
+	var (
+		engines       = make([]*promql.Engine, len(deltas))
+		defaultEngine = newEngine(lookbackDelta)
+		ld            = lookbackDelta.Milliseconds()
+	)
+	engines[0] = defaultEngine
+	for i, d := range deltas[1:] {
+		if ld < d {
+			// Convert delta from milliseconds to time.Duration (nanoseconds).
+			engines[i+1] = newEngine(time.Duration(d * 1_000_000))
```

Comment:
This looks weird and should fail our lints: `1_000_000` - what about using `timestamp.Time(...)`

Reply:
replaced with standard `time.Duration(d) * time.Millisecond`. Prometheus'es `timestamp` seems unhelpful here, as it's `Duration` we need, not `Time`